lib/src/store.rs

```diff
@@ -168,10 +171,13 @@ impl Store {
 
     pub async fn write_commit(
         self: &Arc<Self>,
-        commit: backend::Commit,
+        mut commit: backend::Commit,
         sign_with: Option<&mut SigningFn<'_>>,
     ) -> BackendResult<Commit> {
         assert!(!commit.parents.is_empty());
+        if !self.record_predecessors {
+            commit.predecessors = vec![];
+        }
```

Comment:
I think this can be handled by `CommitBuilder`, but I don't feel strongly. I assume we'll add `predecessors: Vec` to `CommitBuilder` when we remove it from the `Commit` object.